File: nodejs_websocket_backend/src/data/models/tokens-model.ts
import {TokensEntity} from "../../domain/entity/tokens-entity";

export interface TokensOutputToClient {
    [TokensModel.kAccessToken]: string;
    [TokensModel.kRefreshToken]: string;
    [TokensModel.kUserId]: number;
    [TokensModel.kAccessTokenExpirationMsSinceEpoch]: number;
}

export class TokensModel extends TokensEntity {
    static readonly kAccessToken = "accessToken";
    static readonly kRefreshToken = "refreshToken";
    static readonly kUserId = "userId";
    static readonly kAccessTokenExpirationMsSinceEpoch = "accessTokenExpirationMsSinceEpoch";

    output() : TokensOutputToClient {
        return {
            [TokensModel.kAccessToken]: this.accessToken,
            [TokensModel.kRefreshToken]: this.refreshToken,
            [TokensModel.kUserId]: this.userId,
            [TokensModel.kAccessTokenExpirationMsSinceEpoch]: this.accessTokenExpiration.getTime(),
        }
    }

    static fromEntity (entity:TokensEntity) : TokensModel {
        return Object.assign(new TokensModel(null,null,null,null), entity);
    }

}

File: flutter_app/lib/features/login_and_registration/data/models/tokens_model.dart
import '../../domain/entities/tokens_entity.dart';



class TokensModel extends TokensEntity {
  static const _kAccessToken = "accessToken";
  static const _kRefreshToken = "refreshToken";
  static const _kUserId = "userId";
  static const _kAccessTokenExpirationMsSinceEpoch = "accessTokenExpirationMsSinceEpoch";

  TokensModel({
    required String accessToken,
    required String refreshToken,
    required int loggedUserId,
    required DateTime accessTokenExpiration
  }) : super (
    userId: loggedUserId,
    accessToken: accessToken,
    refreshToken: refreshToken,
    accessTokenExpiration: accessTokenExpiration,
  );

  static TokensModel fromMap (output) {
    return TokensModel(
      accessToken: output[_kAccessToken],
      refreshToken: output[_kRefreshToken],
      loggedUserId: output[_kUserId],
      accessTokenExpiration: DateTime.fromMillisecondsSinceEpoch(output[_kAccessTokenExpirationMsSinceEpoch])
    );
  }

}